csrc/executor.cpp

```diff
@@ -1835,12 +1835,23 @@ float FusionExecutor::runRtc(
   std::vector<void*> pointers;
 
   for (const auto& input : args) {
+    StructOf dtype;
+    dtype.field_names = {"data", "logical_size", "alloc_stride"};
+    dtype.types["data"] = NVFUSER_MAYBE_MAKE_SHARED(PointerOf{
+        std::make_shared<DataType>(aten_to_data_type(input.scalar_type()))});
+    dtype.types["logical_size"] = NVFUSER_MAYBE_MAKE_SHARED2(ArrayOf{
+        std::make_shared<DataType>(DataType::Index), (size_t)input.dim()});
+    dtype.types["alloc_stride"] = NVFUSER_MAYBE_MAKE_SHARED2(ArrayOf{
+        std::make_shared<DataType>(DataType::Index), (size_t)input.dim()});
+
```

Comment:
So this is the `PolymorphicValue` dtype corresponding to a `TensorView` with this size and dtype. It seems like this pattern may be re-used often as something like `tensorViewDescriptor(ndim, dtype)`. It's a little confusing to me to read this since the term "dtype" plays two roles here: the input tensor has a scalar dtype that maps to a `PrimDataType`. But here the `dtype` variable is a `DataType` used by `PolymorphicValue` that can hold this descriptor of a `TensorView`. I don't see any way to improve that; I'll just need to get used to `DataType` describing things that can themselves have `DataType`s,

Reply:
Added a `globalTensorMetaData`.

> I'll just need to get used to `DataType` describing things that can themselves have `DataType`s

I think this is pretty common in programming languages. For example, there are "pointer of float", "pointer of pointer of pointer of pointer of a function that returns a pointer of pointer of float and takes an array of pointer of float and a struct with a pointer of float and a int fields".